auslib/test/admin/views/test_rules.py

```diff
@@ -594,3 +623,335 @@ def testGetRules(self):
+            "telemetry_product": "foo", "telemetry_channel": "bar", "telemetry_uptake": 42, "scheduled_by": "bill", "base_rule_id": 5,
+            "base_priority": 80, "base_buildTarget": "d", "base_version": "3.3", "base_backgroundRate": 100, "base_mapping": "c", "base_update_type": "minor",
+            "base_data_version": 1, "data_version": 1, "sc_id": 5, "when": None, "complete": False, "base_alias": None, "base_product": None,
+            "base_channel": None, "base_buildID": None, "base_locale": None, "base_osVersion": None, "base_distribution": None, "base_distVersion": None,
+            "base_headerArchitecture": None, "base_comment": None, "base_whitelist": None, "base_systemCapabilities": None,
+        }
+        self.assertEquals(db_data, expected)
+
+    def testAddScheduledChangeNewRule(self):
+        data = {
+            "when": 1234567, "priority": 120, "backgroundRate": 100, "product": "blah", "channel": "blah",
+            "update_type": "minor", "mapping": "a",
+        }
+        ret = self._post("/scheduled_changes/rules", data=data)
+        self.assertEquals(ret.status_code, 200, ret.data)
+        self.assertEquals(json.loads(ret.data), {"sc_id": 5})
+
+        r = dbo.rules.scheduled_changes.t.select().where(dbo.rules.scheduled_changes.sc_id == 5).execute().fetchall()
+        self.assertEquals(len(r), 1)
+        db_data = dict(r[0])
+        expected = {
+            "when": 1234567, "scheduled_by": "bill", "base_priority": 120, "base_backgroundRate": 100, "base_product": "blah", "base_channel": "blah",
+            "base_update_type": "minor", "base_mapping": "a", "sc_id": 5, "data_version": 1, "complete": False, "base_data_version": None,
+            "telemetry_product": None, "telemetry_channel": None, "telemetry_uptake": None, "base_rule_id": None, "base_buildTarget": None,
+            "base_version": None, "base_alias": None, "base_buildID": None, "base_locale": None, "base_osVersion": None, "base_distribution": None,
+            "base_distVersion": None, "base_headerArchitecture": None, "base_comment": None, "base_whitelist": None, "base_systemCapabilities": None,
+        }
+        self.assertEquals(db_data, expected)
+
+    def testAddScheduledChangeNoPermissionsToSchedule(self):
+        data = {
+            "when": 1234567, "priority": 120, "backgroundRate": 100, "product": "blah", "channel": "blah",
+            "update_type": "minor", "mapping": "a",
+        }
+        ret = self._post("/scheduled_changes/rules", data=data, username="bob")
+        self.assertEquals(ret.status_code, 403, ret.data)
+
+    def testAddScheduledChangeNoPermissionsToMakeChange(self):
+        data = {
+            "when": 1234567, "priority": 120, "backgroundRate": 100, "product": "foo", "channel": "blah",
+            "update_type": "minor", "mapping": "a",
+        }
+        ret = self._post("/scheduled_changes/rules", data=data, username="mary")
+        self.assertEquals(ret.status_code, 403, ret.data)
```

Comment:
IIRC 'mary' is allowed to enact changes but not change /rules.  Are you specifically testing this combination or could it be any user without /rules perm ? If the latter it would be clearer to use something else.